packages/grid/x-data-grid/src/hooks/features/pagination/gridPaginationInterfaces.ts

```diff
@@ -1,38 +1,32 @@
+import { GridPaginationModel } from '../../../models/gridPaginationProps';
+
 export interface GridPaginationState {
-  pageSize: number;
-  page: number;
+  paginationModel: GridPaginationModel;
   pageCount: number;
   rowCount: number;
 }
 
 export interface GridPaginationInitialState {
-  pageSize?: number;
-  page?: number;
+  paginationModel?: Partial<GridPaginationModel>;
```

Comment:
Do we need a state structure like:
```
{
  paginationModel: {
    page:
    pageSize:
  }
}

Can't we make the internal state a single level?

```suggestion
  export interface GridPaginationInitialState extends Partial<GridPaginationModel> {};
```

Reply:
> Can't we make the internal state a single level?

You mean `InitialState` or state itself. I observed a bit the patterns for all other controlled states. I feel if we make it single level it will make things a bit inconsistent.

Initial State:

For instance in row grouping:
https://github.com/mui/mui-x/blob/00d4569c716fcda032a550b5ab15ea22d0a17c37/packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/gridRowGroupingInterfaces.ts#L9-L11

For aggregation:
https://github.com/mui/mui-x/blob/00d4569c716fcda032a550b5ab15ea22d0a17c37/packages/grid/x-data-grid-premium/src/hooks/features/aggregation/gridAggregationInterfaces.ts#L13-L15

Same is the case with sorting and filtering, although a small inconsistency I can find is different keywords used for them i.e `filterModel` and `sortModel` instead of `model` as in aggregation and rowGrouping. May be we can think about making it consistent everywhere?

By the way, does making single level you mean to make the prop interface like the following or only refactoring internally?

```diff
<DataGrid
  initialState={
    pagination: {
-     paginationModel: {
      page: initialPage,
      pageSize: initialPageSize
-     }
    }
  }
/>
```

Reply:
`sorting` and `filter` are other examples of nested states. Both contain only the model.
I think we should keep the pagination state consistent with those states - see https://github.com/mui/mui-x/pull/7147#discussion_r1069682438

Reply:
If we start exploring in this direction then another option is to do something like this:
```diff
<DataGrid
  initialState={
-    pagination: {
+     paginationModel: {
      page: initialPage,
      pageSize: initialPageSize
     }
-   }
  }
/>
```

That is leaning on @m4theushw comment about the pagination being a 2-year-old feature and we never needed more than a model in the state. I guess we are short on time for this discussion so we can have this in mind for the future but for now, we can keep it as is.